### PR TITLE
Fixes bug with wrong forms being available to the user

### PR DESCRIPTION
**Pull request by sophiabersani**
https://www.pivotaltracker.com/story/show/120448065
https://www.pivotaltracker.com/story/show/120457067
- Fixes cached forms ignoring submissions and getting re-displayed as a result.
- Fixes an issue with the back button after submitting a response.

### CODE REVIEW

File: js/api/Forms.js [hunk 1]
```diff
@@ -37,7 +37,27 @@ export function loadCachedFormDataById(formId) {
 
 // Fetches the cached forms related to a specific survey
 export function loadCachedForms(surveyId) {
-  return realm.objects('Form').filtered(`surveyId = "${surveyId}"`)
+  let forms = realm.objects('Form').filtered(`surveyId = "${surveyId}"`)
+  let submissions = realm.objects('Submission')
+  let formList = []
+
+  // Switch out from Realm list to standard array
```
Comment (dan-nyanko) [inline js/api/Forms.js:44]: why create an array and not iterate directly over the realm result?

Comment (sophiabersani) [inline js/api/Forms.js:44]: Unless I am missing something, realm result is an object and not an array. 
I found excluding/removing target items from it a bit unreliable, as I'd end up with holes in the middle of the list.

Comment (dan-nyanko) [inline js/api/Forms.js:44]: https://realm.io/docs/react-native/latest/api/Realm.Collection.html

I think its an instance of Realm.Collection, which has an .entries() method as the iterable interface

Comment (sophiabersani) [inline js/api/Forms.js:44]: You're right. My issues might have come from a bad interaction with lodash.

**Comment by sophiabersani** [comment]
I will close this PR for now as this bugfix is incompatible with #69.
That PR takes care of most issues that were causing this to happen, so I am keeping an eye on this behavior and will resubmit the problem re-appears.

Making a tiny unobstructive PR to take care of the other issue.

**Pull request closed without merging**